---
title: "Research"
layout: textlay
excerpt: "Research"
sitemap: false
permalink: /research/
---


Research topics
---------------

**We carry out research on the theory, methods and applications of
modern statistical data analysis. Our focus is on data-rich
application fields including microbial ecology, functional genomics,
and computational history.** There is a great demand for targeted
algorithmic methods for extracting information from vast data
collections with minimal human intervention. By combining information
across multiple, complementary sources it is possible to overcome some
of the limitations and statistical uncertainties associated with
individual data sets. Open source implementations turn the new models
into accessible tools to guide modeling and experimentation.

![]({{ site.url }}{{ site.baseurl }}/images/isme14.jpg){: style="width: 300px; float: right; border: 20px; margin-left: 30px;"}

**Modern statistical data analysis** Our research blends elements from multiple theoretical and methodological fields, including statistical machine learning/AI, probabilistic programming, numerical ecology, and data science.

**Human microbiomics and statistical ecology** Human gut microbiota constitutes one of the most densely populated ecosystem on our planet. The known pan-genome
carries 2-3 orders of magnitude more unique genes than our own genome,
it is highly dynamic in space and time, and a key contributor to
immune system, digestion of food, and various health
complications. Characterizing the overall structure, variability, and
health associations of this virtual metabolic organ forms a major
challenge for contemporary human biology. We have contributed to
research on some of the largest existing population studies of this
microbial ecosystem, mapping large-scale associations with [diet,
lifestyle, physiology, and
well-being](http://doi.org/10.1038/ncomms7342), developed the [tipping
elements](http://doi.org/10.1038/ncomms5344) concept, and investigated
the [mechanisms](http://doi.org/10.1109/JPROC.2015.2428213) and
[dynamics](http://doi.org/10.1016/j.mib.2015.04.004) of microbial
community assembly.

<!--In [Department of
Veterinary Bioscience, University of
Helsinki](http://www.vetmed.helsinki.fi/english/vetbio/index.html) and
[Laboratory of Microbiology, Wageningen
University](http://www.mib.wur.nl/UK/), we work to understand the
dynamics of the human gut ecosystem based on high-throughput microbial
profiling data {[Poster at ISMB](http://posters.f1000.com/P1886)}.-->

![]({{ site.url }}{{ site.baseurl }}/images/dialister.png){: style="width: 300px; float: left; border: 10px"}

**Functional genomics** Mapping of the three billion base-pair human
genome sequence in 2001 was the first step towards uncovering the
dynamic and contextual functional properties of the
genome. Understanding functional organization of genetic information
and its regulation through transcriptional, epigenetic, and other
mechanisms remains a key challenge for human biology. We have done
research in cancer studies, large gene expression databases, and
multi-omics data integration.  <!--In the thesis work, I developed
open source machine learning algorithms for data integration in
functional genomics.  For more details, check the [list of
publications](/work/pubs) and the [PhD
thesis](http://lib.tkk.fi/Diss/2010/isbn9789526033686).-->

**Computational humanities** We develop open algorithms and research
tools for digital humanities as part of [Helsinki Computational
History (COMHIS) Group](http://comhis.github.io) and as the founder of
[rOpenGov](https://ropengov.github.io). Our current focus is in
modeling the history of public discourse across several centuries
following the emergence of print press in the 15th century (see a [poster in Digital Humanities 2019](https://www.helsinki.fi/en/researchgroups/computational-history/dh2019-poster)). We also publish algorithms open government data, such as [eurostat](http://ropengov.github.io/eurostat/) and manage the [rOpenGov](https://ropengov.github.io) developer network, which has over 20 R packages and tens of thousands of annual downloads.

**Open science** We have actively supported [Open Knowledge Finland
  Open Science work group](https://fi.okfn.org/wg/openscience/), which
  received **Open Science and Research award of the Ministry of
  Education and Culture in 2017**. Some highlights include a [Report on
  the openness of academic
  publishers](https://openscience.fi/-/openness-of-academic-publishers-evaluated-for-the-first-time),
  commissioned by Finnish Ministry of Education and Culture; [opening of several agreements with academic publishers in 2017-2019](http://finelib.fi/finelib-agreements-published/) and [opening
  of scientific journal subscription costs in Finland
  2010-2017](https://fi.okfn.org/2018/08/27/finland-remains-a-leading-country-in-the-transparency-of-academic-publishing-costs)
  following our FOI requests.





<!--
**Teaching**  
See [separate page](/work/teaching)
-->

<!--**Past**  
[A brief biography of my research career](/info/bio). 
-->

<!--

**Key collaborators**

- [Jeroen Raes Lab](http://www.raeslab.org/) at VIB/KU Leuven on bioinformatics and ecosystems< biology.
- [Prof Willem de Vos](https://tuhat.helsinki.fi/portal/fi/person/wdevos) Helsinki/Wageningen; human gut microbial ecology. 
- Prof Mikko Tolonen. I am an active and founding member of [Helsinki Computational History Group](http://comhis.github.io/)
-->


**We are grateful for support from our collaborators and research funders.**

<figure class="fourth">
  <img src="{{ site.url }}{{ site.baseurl }}/images/logopic/aka.png" style="width: 100px">
  <img src="{{ site.url }}{{ site.baseurl }}/images/logopic/turun_yliopisto_logo_rgb.png" style="width: 90px">
  <img src="{{ site.url }}{{ site.baseurl }}/images/logopic/sitra.png" style="width: 110px">
  <img src="{{ site.url }}{{ site.baseurl }}/images/logopic/cost.png" style="width: 110px">
  <img src="{{ site.url }}{{ site.baseurl }}/images/logopic/kordelin.png" style="width: 70px">
  <img src="{{ site.url }}{{ site.baseurl }}/images/logopic/hri.png" style="width: 90px">
  <img src="{{ site.url }}{{ site.baseurl }}/images/logopic/apps4-logo.jpg" style="width: 100px">
  <img src="{{ site.url }}{{ site.baseurl }}/images/logopic/wici.png" style="width: 110px">
  <img src="{{ site.url }}{{ site.baseurl }}/images/logopic/dupuytren.jpg" style="width: 80px">
  <img src="{{ site.url }}{{ site.baseurl }}/images/logopic/Blueprint1.png" style="width: 70px">              
</figure>




 
